### PR TITLE
Fix broken cl edit function

### LINKED ISSUE #2237: [USER ISSUE] cl edit doesn't work
**Describe the bug**
When I try to use `cl edit` from the command line I get an exception

**Codalab Information**
- *Your username*: robinjia
- *UUID of worksheet where problem occurred*: 0x23acc0bd2aa044fd9300f73d3a5fbbb9
- *UUID of problem bundle (please don't delete)*: n/a
- *Error message received from Codalab (if applicable)*:

```
Traceback (most recent call last):
  File "/home/robinjia/.local/bin/cl", line 8, in <module>
    sys.exit(main())
  File "/home/robinjia/.local/lib/python3.6/site-packages/codalab/bin/cl.py", line 10, in main
    cli.do_command(sys.argv[1:])
  File "/home/robinjia/.local/lib/python3.6/site-packages/codalab/lib/bundle_cli.py", line 864, in do_command
    structured_result = command_fn()
  File "/home/robinjia/.local/lib/python3.6/site-packages/codalab/lib/bundle_cli.py", line 858, in <lambda>
    command_fn = lambda: args.function(self, args)
  File "/home/robinjia/.local/lib/python3.6/site-packages/codalab/lib/bundle_cli.py", line 1708, in do_edit_command
    bundle_subclass, info['metadata']
  File "/home/robinjia/.local/lib/python3.6/site-packages/codalab/lib/metadata_util.py", line 73, in request_missing_metadata
    return parse_metadata_form(bundle_subclass, form_result)
  File "/home/robinjia/.local/lib/python3.6/site-packages/codalab/lib/metadata_util.py", line 85, in parse_metadata_form
    if line != '' and not line.startswith('//'):
TypeError: startswith first arg must be bytes or a tuple of bytes, not str
```

 - *OS (if on CLI)*: Ubuntu 18.04
 - *Codalab CLI Version (if on CLI)*: 0.5.13
 - *Browser (if on Web)*;

**To Reproduce**
Steps to reproduce the behavior :
1. `cl edit ^1` or it seems any bundle (mine opens vim)
2. Save and close

**Expected behavior**
Bundle info should update with no exception thrown

### PR DESCRIPTION
**Pull request by candicegjing**
Fixed #2237 

This PR will address the following two issues:
1. The original problem of `cl edit` reported in #2237 
1. Another problem is introduced from an old PR which added the functionality to delete `staged_status` using `None` value. Using `None` value to indicate deletion operation seems to be conflicting with existing features (e.g. `cl edit`). I added a new parameter to indicate the deletion operation for now. But I am open to any other suggestions.

### CODE REVIEW

File: codalab/lib/metadata_util.py [hunk 1]
```diff
@@ -94,13 +94,14 @@ def parse_metadata_form(bundle_subclass, form_result):
...
-                if any(unicode_util.contains_unicode(v) for v in result[metadata_key]):
+                remainders = remainder.split() if remainder else []
+                if any(unicode_util.contains_unicode(r) for r in remainders):
                     raise UsageError(
-                        'Metadata cannot contain unicode: %s = %s'
-                        % (metadata_key, result[metadata_key])
+                        'Metadata cannot contain unicode: %s = %s' % (metadata_key, remainder)
                     )
-                result[metadata_key] = remainder.split() if remainder else []
+                result[metadata_key] = remainders
```
Comment (candicegjing) [inline codalab/lib/metadata_util.py:104]: Not exactly sure if I have made the right changes here. From the context, it looks like we want to check for each element of the remainder to see if any of it contains Unicode character. But let me know if this is not the case.

File: codalab/server/bundle_manager.py [hunk 2]
```diff
@@ -826,7 +826,9 @@ def _get_staged_bundles_to_run(self, workers, user_info_cache):
                     # Remove the uuid from self._bundles_without_matched_workers if a matched
                     # private worker is found in the system and update bundle's metadata
                     if bundle.uuid in self._bundles_without_matched_workers:
-                        self._model.update_bundle(bundle, {'metadata': {'staged_status': None}})
+                        self._model.update_bundle(
```
Comment (percyliang) [inline codalab/server/bundle_manager.py:851]: Why is this related to `cl edit`?  Should be in a different PR?

Comment (candicegjing) [reply codalab/server/bundle_manager.py:851]: `cl edit` doesn't work properly with the above string decoding fix. When a user types `cl edit <uuid>` to edit metadata fields in the pop-up window, if any of the fields are empty, CLI will just return a metadata key is missing as follows, as the following field has value `None`, which was treated as a deletion operation in update function.
```
stanford@Stanfords-MacBook-Pro codalab-worksheets % cl edit 0x075eaa877bf84ea68072995b19891474  
UsageError: Missing metadata key: allow_failed_dependencies
```

File: codalab/model/bundle_model.py [hunk 3]
```diff
@@ -1106,7 +1106,8 @@ def update_bundle(self, bundle, update, connection=None):
         # Generate a list of metadata keys that will be deleted and udpate metadata key-value pair
         metadata_delete_keys = []
         for key, value in metadata_update.items():
-            if value is None:
+            # Delete the key,value pair when the delete flag is set to True and value is None
+            if delete and value is None:
```
Comment (percyliang) [inline codalab/model/bundle_model.py:1112]: We are changing the default behavior from `delete=True` to `delete=False`...are there any other calls to `update_bundle` that we need to update?

**Review by percyliang — APPROVED** [review]

File: codalab/model/bundle_model.py (hunk 3 above)
Comment (candicegjing) [reply codalab/model/bundle_model.py:1112]: No... `delete` operation is only introduced when we were trying to add the metadata field `staged_status`, as this field will disappear once the bundle successfully gets assigned to a worker. We didn't have this before, so no other functions or features are affected.

Comment (percyliang) [reply codalab/model/bundle_model.py:1112]: The previous code is equivalent to `delete=True`, but now the default is `delete=False`.
